File: app/src/main/java/com/ninjahoahong/unstoppable/AppComponent.java
package com.ninjahoahong.unstoppable;

import android.content.Context;

import com.ninjahoahong.unstoppable.data.source.QuestionRepository;
import com.ninjahoahong.unstoppable.data.source.ScheduleRepositoryModule;
import com.ninjahoahong.unstoppable.data.source.local.LocalDataSource;
import com.ninjahoahong.unstoppable.data.source.remote.RemoteDataSource;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ScheduleRepositoryModule.class, AppModule.class, NetModule.class})
public interface AppComponent {

    void inject(RemoteDataSource remoteDataSource);
    void inject(LocalDataSource localDataSource);
    QuestionRepository getScheduleRepository();
    Context getContext();
}

File: app/src/main/java/com/ninjahoahong/unstoppable/AppComponent.kt
package com.ninjahoahong.unstoppable

import com.ninjahoahong.unstoppable.question.QuestionViewFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    (AppModule::class)
])
interface AppComponent {
    fun inject(questionViewFragment: QuestionViewFragment)
}